# Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
# Patch for OpenModelica 1.13.2 (https://www.openmodelica.org)
# 
# All rights reserved.
# 
# THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
# THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
# ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES RECIPIENT'S ACCEPTANCE
# OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3, ACCORDING TO RECIPIENTS CHOICE.
# 
# The OpenModelica software and the Open Source Modelica
# Consortium (OSMC) Public License (OSMC-PL) are obtained
# from OSMC, either from the above address,
# from the URLs: http://www.ida.liu.se/projects/OpenModelica or
# http://www.openmodelica.org, and in the OpenModelica distribution.
# GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
# 
# This program is distributed WITHOUT ANY WARRANTY; without
# even the implied warranty of  MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
# IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
# 
# See the full OSMC Public License conditions for more details.
# SPDX-License-Identifier: GPL-3.0 OR LicenseRef-OSMC-PL
Index: OMCompiler/Compiler/FrontEnd/ModelicaBuiltin.mo
===================================================================
--- OMCompiler/Compiler/FrontEnd/ModelicaBuiltin.mo.orig	2019-04-11 11:31:10.417525274 +0200
+++ OMCompiler/Compiler/FrontEnd/ModelicaBuiltin.mo	2019-04-11 11:31:10.462525659 +0200
@@ -2165,6 +2165,16 @@ annotation(Documentation(info="<html>
 </html>"),preferredView="text");
 end dumpXMLDAE;
 
+function createXML2RTE "Outputs the DAE system corresponding to a specific RTE model"
+   input TypeName className;
+   input String   rewriteRulesFile = ""  "The file from where the rewiteRules are read, default is empty which means no rewrite rules";
+
+   output Boolean success    "If the function succeeded true/false";
+   output String xmlfileName "The output specific XML file";
+external "builtin";
+annotation(preferredView="text");
+end createXML2RTE;
+
 function convertUnits
   input String s1;
   input String s2;
